### PR TITLE
Use automatic_lr_schedule var of lightning module let user decide whether to call lr_scheduler.step() automatically in manual optimization mode.

### LINKED ISSUE #12284: Let user decide whether to call lr_scheduler.step() automatically in manual optimization mode.
## 🚀 Feature

Like `self.automatic_optimization` add a var name `automatic_lr_schedule` or `automatic_lr_schedule_when_manual_optimization` in LightningModule with default value False, which User can set it to True.

When it is set to False, everything will run as now, but when it is set to True and the module is running in manaul_optimization mode, pytorch lightning will automatically call lr_scheduler.step() as in automatic optimization mode.

### Motivation

See [here](https://pytorch-lightning.readthedocs.io/en/stable/common/optimizers.html#learning-rate-scheduling-manual).

> Before 1.3, Lightning automatically called lr_scheduler.step() in both automatic and manual optimization. From 1.3, lr_scheduler.step() is now for the user to call at arbitrary intervals.

In my use case(like train a GAN), I surely have to call `optimizer.zero_grad()`, `self.manual_backward()` and `optimizer.step()` by myself, but for lr_schedulers, what I want to do is just like things done by pytorch lightning in automatic_optimization mode.

Therefore, I want to add this flag to make pytorch lightning automatically call lr_scheduler.step() for my module instead of write the logical code by myself.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @akihironitta

### PR DESCRIPTION
**Pull request by shenmishajing**
## What does this PR do?

Fixes #12284

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by shenmishajing** [comment]
close this as [here](https://github.com/PyTorchLightning/pytorch-lightning/issues/12284#issuecomment-1069064740).
